File: 🟡 Medium/2707 - Extra Characters in a String/_2707ExtraCharactersInAString.cpp
class Solution {
public:
    int longestCommonPrefix(vector<int>& arr1, vector<int>& arr2) {
        int res = 0;

        unordered_set<int> prefixesOfarr1;

        for (auto& el : arr1) {
            while (!prefixesOfarr1.count(el) && el > 0) {
                prefixesOfarr1.insert(el);
                el /= 10;
            }
        }

        for (auto& el : arr2) {
            while (!prefixesOfarr1.count(el) && el > 0) {
                el /= 10;
            }

            if (el > 0) {
                res = max(res, (int)floor(log10(el) + 1));
            }
        }

        return res;
    }
};

File: 🟡 Medium/2707 - Extra Characters in a String/_2707ExtraCharactersInAString.java
class Solution {
    public int longestCommonPrefix(int[] arr1, int[] arr2) {
        int res = 0;

        Set<Integer> prefixesOfarr1 = new HashSet<>();

        for (int el : arr1) {
            while (!prefixesOfarr1.contains(el) && el > 0) {
                prefixesOfarr1.add(el);
                el /= 10;
            }
        }

        for (int el : arr2) {
            while (!prefixesOfarr1.contains(el) && el > 0) {
                el /= 10;
            }

            if (el > 0) {
                res = Math.max(res, (int) Math.floor(Math.log10(el) + 1));
            }
        }

        return res;
    }
}